api/src/main/java/org/apache/iceberg/types/ImportCompatibilityChecker.java

```diff
@@ -0,0 +1,149 @@
+/*
```

Comment:
I am trying to see if we can reuse org.apache.iceberg.types.CheckCompatibility#readCompatibilityErrors and avoid this class completely. Will update the PR with my findings